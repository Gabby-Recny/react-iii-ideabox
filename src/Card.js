import React, { useContext } from 'react';
import './Card.css';
import AppContext from './AppContext'

const Card = ({ title, description, id, deleteIdea, isFavorited }) => {
  const [state, dispatch] = useContext(AppContext)

  return (
    <div className={`card ${isFavorited && 'favorite'} ${state.theme}`}>
      <h3>{title}</h3>
      <p>{description}</p>
      <button onClick={() => dispatch({ type: 'REMOVE_IDEA', id })}>ðŸ—‘</button>
    </div>
  )
}

export default Card;
