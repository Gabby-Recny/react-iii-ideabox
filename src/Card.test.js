import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('renders the title of the idea in <h1> tags', () => {
    const wrapper = shallow(<Card title="Bananas" />);
    const title = <h3>Bananas</h3>;

    expect(wrapper.contains(title)).toEqual(true);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<Card title='bananas' description='b a n a n a s' id={3} deleteIdea={jest.fn()} />)

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the favorited snapshot', () => {
    const wrapper = shallow(<Card title='bananas' description='b a n a n a s' id={3} deleteIdea={jest.fn()} isFavorited={true} />)

    expect(wrapper).toMatchSnapshot();
  });

  it('should call the deleteIdea prop when clicked', () => {
    const deleteIdeaMock = jest.fn();

    const wrapper = shallow(
      <Card
        title="Bananas"
        description="blah blah bloop"
        id={7}
        deleteIdea={deleteIdeaMock}
      />
    );

    wrapper.find('button').simulate('click');

    expect(deleteIdeaMock).toBeCalledWith(7);
  });
});
