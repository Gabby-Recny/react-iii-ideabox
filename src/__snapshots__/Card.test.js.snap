// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the favorited snapshot 1`] = `
<div
  className="card favorite"
>
  <h3>
    bananas
  </h3>
  <p>
    b a n a n a s
  </p>
  <button
    onClick={[Function]}
  >
    ðŸ—‘
  </button>
</div>
`;

exports[`Card should match the snapshot 1`] = `
<div
  className="card undefined"
>
  <h3>
    bananas
  </h3>
  <p>
    b a n a n a s
  </p>
  <button
    onClick={[Function]}
  >
    ðŸ—‘
  </button>
</div>
`;
